metrics/opentelemetry.go

```diff
@@ -0,0 +1,226 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/goto/meteor/agent"
+	"github.com/goto/meteor/config"
+	"github.com/goto/meteor/recipe"
+	"github.com/goto/salt/log"
+	"go.opentelemetry.io/contrib/instrumentation/host"
+	"go.opentelemetry.io/contrib/instrumentation/runtime"
+	"go.opentelemetry.io/contrib/samplers/probability/consistent"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/propagation"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/encoding/gzip"
+)
+
+const gracePeriod = 5 * time.Second
+
+// OtelMonitor represents the otel monitor.
+type OtelMonitor struct {
```

Comment:
Can we split the code for `OtelMonitor` into a different file from this one?

Reply:
updated